lib/improver/nowcasting/optical_flow.py

```diff
@@ -932,13 +946,13 @@ def process(self, cube1, cube2, boxsize=30):
         t_coord = cube2.coord("time")
 
         ucube = iris.cube.Cube(
-            ucomp, long_name="advection_velocity_x", units="m s-1",
-            dim_coords_and_dims=[(y_coord, 0), (x_coord, 1)])
+            ucomp, long_name="precipitation_advection_x_velocity",
```

Comment:
This is assuming that the field is precipitation but there is nothing anywhere else that checks that this is the case. Please can we add a check and throw a warning (including that the parameters have been tuned for precipitation fields and may not work correctly with other parameters).

Reply:
Good point, will do.